packages/edit-site/src/components/add-new-template/new-template.js

```diff
@@ -266,3 +192,91 @@ export default function NewTemplate( { postType } ) {
 		</>
 	);
 }
+
+function useMissingTemplates(
+	setEntityForSuggestions,
+	setShowCustomTemplateModal
+) {
+	const postTypes = usePostTypes();
+	const taxonomies = useTaxonomies();
+	const categoryTaxonomy = useTaxonomyCategory();
+	const tagTaxonomy = useTaxonomyTag();
+
+	const existingTemplates = useExistingTemplates();
+	const defaultTemplateTypes = useDefaultTemplateTypes();
+
+	const existingTemplateSlugs = ( existingTemplates || [] ).map(
+		( { slug } ) => slug
+	);
+
+	const missingDefaultTemplates = ( defaultTemplateTypes || [] ).filter(
+		( template ) =>
+			DEFAULT_TEMPLATE_SLUGS.includes( template.slug ) &&
+			! existingTemplateSlugs.includes( template.slug )
+	);
+	const onClickMenuItem = ( _entityForSuggestions ) => {
+		setShowCustomTemplateModal( true );
+		setEntityForSuggestions( _entityForSuggestions );
+	};
+	// TODO: find better names for many new vars :)
```

Comment:
The `TODO` is acknowledged, and we will address it in the context of https://github.com/WordPress/gutenberg/pull/41875#issuecomment-1172561934.